src/lib.rs

```diff
@@ -2096,6 +2096,38 @@ impl Build {
             tool
         };
 
+        // New "standalone" C/C++ cross-compiler executables from recent Android NDK
+        // are just shell scripts that call main clang binary (from Android NDK) with
+        // proper `--target` argument.
+        //
+        // For example, armv7a-linux-androideabi16-clang passes
+        // `--target=armv7a-linux-androideabi16` to clang.
+        //
+        // As the shell script calls the main clang binary, the command line limit length
+        // on Windows is restricted to around 8k characters instead of around 32k characters.
+        // To remove this limit, we call the main clang binary directly and contruct the
+        // `--target=` ourselves.
+        if host.contains("windows") && android_clang_compiler_uses_target_arg_internally(&tool.path)
+        {
+            let mut f = || -> Option<bool> {
+                let file_name = tool.path.file_name()?.to_str()?.to_owned();
+                let (target, clang) = file_name.split_at(file_name.rfind("-")?);
+
+                tool.path.set_file_name(clang.trim_start_matches("-"));
+                tool.path.set_extension("exe");
+                tool.args.push(format!("--target={}", target).into());
+
+                for version in 16..=23 {
+                    if target.contains(&format!("i686-linux-android{}", version)) {
+                        tool.args.push("-mstackrealign".into());
```

Comment:
Is this what the current scripts do? Is there a reason to perhaps not pass this unconditionally? (I have no idea what this flag does)

Reply:
Yea that's what the scripts do. Only the `i686-linux-android16-23` have the `-mstackrealign` option added in the script, which force realigns the stack at entry of every function. Don't think we want to do that unconditionally as I am not sure what the flag does as well.

Reply:
Hm ok, can you add a comment for why this arg is added? Also, is 23 the latest version? Or do newer versions not require this argument?

Reply:
Newer versions do not require the arguments

Reply:
Hm ok, in that case could this perhaps be tweaked a bit differently to instead of looping over versions to instead take a look at the target and try to parse a version at the end? If this no longer happens then this is basically just handling older historical compilers then.

Reply:
Done!